.github/workflows/build-and-publish.yml

```diff
@@ -48,8 +51,12 @@ jobs:
       with:
         images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
         flavor: |
-          latest=${{ github.event_name == 'push' }}
-        tags: ${{ github.ref_name }}
+          latest=${{ github.ref == 'refs/heads/master' }}
+        tags: |
+          type=raw,value=latest,enable=${{ github.ref == 'refs/heads/master' }}
+          type=edge,value=edge,enable=${{ startsWith(github.ref, 'refs/heads/upcoming/') }}
+          type=raw,value=upcoming,enable=${{ startsWith(github.ref, 'refs/heads/upcoming/') }}
+          type=ref,event=tag
```

Comment:
What does `type=`, `value=` and `event=` do?

Reply:
`type` tells docker the type of image (determines the tag type)
`value` tells docker the value for the tag (`value=edge`) means the tag will be `:edge`

  * raw = tag is set to whatever is set for value, so `type=raw,value=latest` will produce an image with the tag `:latest`
  * edge = Image Tag is set to `:edge`, edge also shows up in the registries website as unstable iirc. 
  * ref = Sets tag to the reference for the case of `type=ref,event=tag` it uses the "tag Github event" as a reference so setting a GH Tag of `v1.1` will set a image tag of `:v1.1`